bodhi/tests/server/base.py

```diff
@@ -19,70 +19,198 @@
+    """
+    Creates and configures a test database for Bodhi.
+
+    .. note::
+        For some reason, this fails on the in-memory version of SQLite with an error
+        about nested transactions.
 
-FAITOUT = 'http://209.132.184.152/faitout/'
-# The BUILD_ID environment variable is set by Jenkins and allows us to detect if
-# we are running the tests in jenkins or not
-# https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-below
-if os.environ.get('BUILD_ID'):
-    try:
-        req = requests.get('%s/new' % FAITOUT, timeout=60)
-        if req.status_code == 200:
-            DB_PATH = req.text
-            DB_NAME = DB_PATH.rsplit('/', 1)[1]
-            print 'Using faitout at: %s' % DB_PATH
-    except:
-        pass
+    Args:
+        db_uri (str): The URI to use when creating the database engine. Defaults to an
+            in-memory SQLite database.
+    """
+    if db_uri.startswith('sqlite:////'):
+        # Clean out any old file
+        db_path = db_uri.split('sqlite:///')[1]
+        if os.path.isfile(db_path):
+            os.unlink(db_path)
+
+    global engine
+    engine = initialize_db({'sqlalchemy.url': db_uri})
+
+    if db_uri.startswith('sqlite://'):
+        # Necessary to get nested transactions working with SQLite. See:
+        # http://docs.sqlalchemy.org/en/latest/dialects/sqlite.html\
+        # #serializable-isolation-savepoints-transactional-ddl
+        @event.listens_for(engine, "connect")
+        def connect_event(dbapi_connection, connection_record):
+            """Stop pysqlite from emitting 'BEGIN'"""
+            # disable pysqlite's emitting of the BEGIN statement entirely.
+            # also stops it from emitting COMMIT before any DDL.
+            dbapi_connection.isolation_level = None
+
+        @event.listens_for(engine, "begin")
+        def begin_event(conn):
+            """Emit our own 'BEGIN' instead of letting pysqlite do it."""
+            conn.execute('BEGIN')
+
+    @event.listens_for(Session, 'after_transaction_end')
+    def restart_savepoint(session, transaction):
+        """Allow tests to call rollback on the session."""
+        if transaction.nested and not transaction._parent.nested:
+            session.expire_all()
+            session.begin_nested()
+
+    return engine
 
 
 class BaseTestCase(unittest.TestCase):
+    """
+    The base test class for Bodhi.
+
+    This class configures the global scoped session with a test database.
+    The test database makes use of nested transactions to provide a clean
+    slate for each test. Tests may call both ``commit`` and ``rollback``
+    on the database session they acquire from ``bodhi.server.Session``.
+    """
     _populate_db = True
 
-    def setUp(self):
+    app_settings = {
+        'authtkt.secret': 'sssshhhhhh',
+        'authtkt.secure': False,
+        'sqlalchemy.url': 'sqlite:////var/tmp/bodhi-tests.sqlite',
```

Comment:
I just copied this from the old functional test class - it turns out it's not necessary and I've removed it.